File: unity/Assets/Scripts/Data/CharacterData.cs
using System.Collections.Generic;
using System.Linq;
using UnityEngine;

namespace Gati.Data
{
    public enum BodyType { SchoolKid, CollegeStudent, DeliveryRider, CricketPlayer, Dancer }

    public class CharacterData
    {
        public string Id;
        public string DisplayName;
        public string Description;
        public BodyType BodyType;
        public Color SkinColor;
        public Color OutfitPrimary;
        public Color OutfitAccent;

        /// Sparks (coins) needed to unlock. 0 = unlocked from the start.
        public int UnlockCost;

        public float MagnetRadiusBonus;
        public float GatiRegenBonus;
    }

    public static class CharacterCatalog
    {
        public static readonly List<CharacterData> All = new List<CharacterData>
        {
            new CharacterData
            {
                Id = "school_kid", DisplayName = "Aarav",
                Description = "School kid racing to beat the first bell. Quick on his feet.",
                BodyType = BodyType.SchoolKid,
                SkinColor = new Color32(0xC9, 0x8A, 0x5A, 255),
                OutfitPrimary = new Color32(0x2C, 0x4E, 0x8A, 255),
                OutfitAccent = new Color32(0xE8, 0xE8, 0xE8, 255),
                UnlockCost = 0, GatiRegenBonus = 0.15f,
            },
            new CharacterData
            {
                Id = "college_student", DisplayName = "Meera",
                Description = "College student with a tote bag full of notes and a bus to catch.",
                BodyType = BodyType.CollegeStudent,
                SkinColor = new Color32(0xA9, 0x70, 0x4A, 255),
                OutfitPrimary = new Color32(0x6B, 0x3F, 0xA0, 255),
                OutfitAccent = new Color32(0xFF, 0xD1, 0x66, 255),
                UnlockCost = 500, MagnetRadiusBonus = 0.2f,
            },
            new CharacterData
            {
                Id = "delivery_rider", DisplayName = "Imran",
                Description = "Delivery rider, orange box on his back, ten minutes on the clock.",
                BodyType = BodyType.DeliveryRider,
                SkinColor = new Color32(0x8A, 0x5A, 0x3B, 255),
                OutfitPrimary = new Color32(0xE8, 0x62, 0x2D, 255),
                OutfitAccent = new Color32(0x1A, 0x1A, 0x1A, 255),
                UnlockCost = 1200, GatiRegenBonus = 0.25f,
            },
            new CharacterData
            {
                Id = "cricket_player", DisplayName = "Rohan",
                Description = "Gully cricket star, bat slung over his shoulder, always sprinting a quick single.",
                BodyType = BodyType.CricketPlayer,
                SkinColor = new Color32(0xB5, 0x7A, 0x4E, 255),
                OutfitPrimary = new Color32(0x1B, 0x7A, 0x3D, 255),
                OutfitAccent = new Color32(0xFF, 0xFF, 0xFF, 255),
                UnlockCost = 2000, MagnetRadiusBonus = 0.35f,
            },
            new CharacterData
            {
                Id = "dancer", DisplayName = "Priya",
                Description = "Classical dancer, anklets jingling, chasing the stage before the curtain rises.",
                BodyType = BodyType.Dancer,
                SkinColor = new Color32(0xCB, 0x8F, 0x5E, 255),
                OutfitPrimary = new Color32(0xC2, 0x2B, 0x5C, 255),
                OutfitAccent = new Color32(0xF6, 0xC4, 0x53, 255),
                UnlockCost = 3000, GatiRegenBonus = 0.2f, MagnetRadiusBonus = 0.2f,
            },
        };

        public static CharacterData ById(string id) =>
            All.FirstOrDefault(c => c.Id == id) ?? All[0];
    }
}

File: lib/game/characters/character_data.dart
import 'package:flutter/material.dart';

/// Body-shape archetype, drawn procedurally by PlayerComponent.
enum BodyType { schoolKid, collegeStudent, deliveryRider, cricketPlayer, dancer }

class CharacterData {
  final String id;
  final String name;
  final String description;
  final BodyType bodyType;
  final Color skinColor;
  final Color outfitPrimary;
  final Color outfitAccent;

  /// Sparks (coins) needed to unlock. 0 = unlocked from the start.
  final int unlockCost;

  /// Small gameplay flavor bonuses — kept subtle so no character feels
  /// mandatory, just a light nudge toward a playstyle.
  final double magnetRadiusBonus;
  final double gatiRegenBonus;

  const CharacterData({
    required this.id,
    required this.name,
    required this.description,
    required this.bodyType,
    required this.skinColor,
    required this.outfitPrimary,
    required this.outfitAccent,
    required this.unlockCost,
    this.magnetRadiusBonus = 0,
    this.gatiRegenBonus = 0,
  });
}

class CharacterCatalog {
  static const List<CharacterData> all = [
    CharacterData(
      id: 'school_kid',
      name: 'Aarav',
      description: 'School kid racing to beat the first bell. Quick on his feet.',
      bodyType: BodyType.schoolKid,
      skinColor: Color(0xFFC98A5A),
      outfitPrimary: Color(0xFF2C4E8A),
      outfitAccent: Color(0xFFE8E8E8),
      unlockCost: 0,
      gatiRegenBonus: 0.15,
    ),
    CharacterData(
      id: 'college_student',
      name: 'Meera',
      description: 'College student with a tote bag full of notes and a bus to catch.',
      bodyType: BodyType.collegeStudent,
      skinColor: Color(0xFFA9704A),
      outfitPrimary: Color(0xFF6B3FA0),
      outfitAccent: Color(0xFFFFD166),
      unlockCost: 500,
      magnetRadiusBonus: 0.2,
    ),
    CharacterData(
      id: 'delivery_rider',
      name: 'Imran',
      description: 'Delivery rider, orange box on his back, ten minutes on the clock.',
      bodyType: BodyType.deliveryRider,
      skinColor: Color(0xFF8A5A3B),
      outfitPrimary: Color(0xFFE8622D),
      outfitAccent: Color(0xFF1A1A1A),
      unlockCost: 1200,
      gatiRegenBonus: 0.25,
    ),
    CharacterData(
      id: 'cricket_player',
      name: 'Rohan',
      description: 'Gully cricket star, bat slung over his shoulder, always sprinting a quick single.',
      bodyType: BodyType.cricketPlayer,
      skinColor: Color(0xFFB57A4E),
      outfitPrimary: Color(0xFF1B7A3D),
      outfitAccent: Color(0xFFFFFFFF),
      unlockCost: 2000,
      magnetRadiusBonus: 0.35,
    ),
    CharacterData(
      id: 'dancer',
      name: 'Priya',
      description: 'Classical dancer, anklets jingling, chasing the stage before the curtain rises.',
      bodyType: BodyType.dancer,
      skinColor: Color(0xFFCB8F5E),
      outfitPrimary: Color(0xFFC22B5C),
      outfitAccent: Color(0xFFF6C453),
      unlockCost: 3000,
      gatiRegenBonus: 0.2,
      magnetRadiusBonus: 0.2,
    ),
  ];

  static CharacterData byId(String id) => all.firstWhere((c) => c.id == id, orElse: () => all.first);
}
